### PR TITLE
feat: improved error handling to improve the error display to the end…

### PR DESCRIPTION
**Pull request by shide1989**
feat: 
- improved error handling to improve the error display to the end user.

### CODE REVIEW

File: src/commands/query.ts [hunk 1]
```diff
@@ -36,13 +36,15 @@ marked.use(markedTerminal() as MarkedExtension);
 const initializeAgentConfig = async (
   workspace: string,
   skipWarmup: boolean
-): Promise<AgentConfig> => {
+): Promise<AgentConfig | null> => {
```
Comment (alxpereira) [inline src/commands/query.ts:39]: @BonnardValentin WDYT ?

File: src/utils/logger.ts [hunk 2]
```diff
@@ -156,8 +156,10 @@ export default class Logger {
     defaultMsg = 'Unexpected error. Please try again !'
   ) {
     if (isDebug) {
+      let msg = '';
       if (axios.isAxiosError(e)) {
         const axiosError = e as AxiosError;
+        msg = (axiosError.response?.data as { error: string })?.error;
```
Comment (alxpereira) [inline src/utils/logger.ts:162]: just do here return `this.cancel((axiosError.response?.data as { error: string })?.error);` after the Logger.error

File: src/utils/logger.ts [hunk 3]
```diff
@@ -172,14 +174,16 @@ export default class Logger {
       } else {
         Logger.error('Command error', e);
       }
-      this.cancel(defaultMsg);
+      this.cancel(msg || defaultMsg);
       return;
     }
 
+    let msg = '';
```
Comment (alxpereira) [inline src/utils/logger.ts:181]: same as above

**Review by alxpereira — DISMISSED** [review]

File: src/utils/logger.ts (hunk 2 above)
Comment (alxpereira) [reply src/utils/logger.ts:162]: no need to have this ugly `let msg`

**Review by BonnardValentin — APPROVED** [review]

File: src/commands/query.ts [hunk 4]
```diff
@@ -95,6 +97,11 @@ export const queryCommand = async (
 
     ////////// Agent Init //////////
     const agentConfig = await initializeAgentConfig(workspace, skipWarmup);
+
+    // If not agent is ellible, it usually means there was an error during the init process that is already displayed.
```
[the remark this answers is not in the document]
Comment (shide1989) [reply src/commands/query.ts:101]: ok ok ...

**Pull request merged**